hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/transfer/ClassicTransfersDecoder.java

```diff
@@ -97,12 +97,19 @@ public TransactionBody decodeCryptoTransfer(
     public TransactionBody decodeCryptoTransferV2(
             @NonNull final byte[] encoded, @NonNull final AddressIdConverter addressIdConverter) {
         final var call = ClassicTransfersTranslator.CRYPTO_TRANSFER_V2.decodeCall(encoded);
-        return bodyOf(tokenTransfers(convertTokenTransfers(
-                        call.get(1),
-                        this::convertingMaybeApprovedAdjustments,
-                        this::convertingMaybeApprovedOwnershipChanges,
-                        addressIdConverter))
-                .transfers(convertingMaybeApprovedAdjustments(((Tuple) call.get(0)).get(0), addressIdConverter)));
+        final var transferList = convertingMaybeApprovedAdjustments(((Tuple) call.get(0)).get(0), addressIdConverter);
+
+        final var cryptoTransfersBody = tokenTransfers(convertTokenTransfers(
+                call.get(1),
+                this::convertingMaybeApprovedAdjustments,
+                this::convertingMaybeApprovedOwnershipChanges,
+                addressIdConverter));
+
+        if (transferList.accountAmounts() != null
```

Comment:
⚠️ Shouldn't it be impossible for this to be `null`? 

Even if _currently_ there's no risk of hiding a bug by working around an incorrect `null` here, adding such checks makes it easier for bugs to hide in the future.

Reply:
Yes, you are right, since in the Builder accountAmounts() method in TransferList, we use List.of() for the input.

I fixed it, great catch!

Reply:
Awesome, thanks @MrValioBg !